models/devices.go

```diff
@@ -0,0 +1,399 @@
+/*
```

Comment:
**Code Review Suggestions**

1. **Consistent Code Formatting**: The code uses a mix of spaces and tabs for indentation. It's recommended to use a consistent indentation using only spaces or only tabs. You can also use tools like `gofmt` to enforce a consistent code formatting.

2. **Unused Imports**: The code imports `github.com/hashicorp/terraform-plugin-framework/diag` but it's not used anywhere. You can remove unused imports to declutter the code.

3. **Error Handling**: The code uses `diag.Diagnostics` but does not handle errors properly. You should add error handling to ensure that the program behaves correctly in case of errors.

4. **Code Comments**: The code has some markdown comments but they are not in a consistent format. You can use a consistent format for comments to make the code easier to read.

5. **Function Length**: The functions are long and do multiple things. You can break them down into smaller functions to make the code easier to read and maintain.

Here is a snippet of the code with some of the suggestions implemented:

```
func newOmeDeviceManagementData(input DeviceManagement) OmeDeviceManagementData {
    // Check for errors before proceeding
    if input.ManagementID <= 0 {
        return OmeDeviceManagementData{}
    }
    a := OmeDeviceManagementData{
        ManagementID:        types.Int64Value(input.ManagementID),
        NetworkAddress:      types.StringValue(input.NetworkAddress.String()),
        MacAddress:          types.StringValue(input.MacAddress),
        ManagementType:      types.Int64Value(input.ManagementType),
        InstrumentationName: types.StringValue(input.InstrumentationName),
        DNSName:             types.StringValue(input.DNSName),
    }
    // Check if the ManagementProfile is not empty
    if input.ManagementProfile != nil {
        mProfs := make([]OmeManagementProfileData, 0)
        for _, mProf := range input.ManagementProfile {
            mProfs = append(mProfs, newOmeManagementProfileData(mProf))
        }
        a.ManagementProfile = mProfs
    }
    return a
}
```

In this example, I have added error checking before proceeding and broken down the function into smaller sections to make it easier to read.